server/src/test/java/org/elasticsearch/gateway/MetaDataStateFormatTests.java

```diff
@@ -329,8 +420,31 @@ private MetaData randomMeta() throws IOException {
 
     private class Format extends MetaDataStateFormat<DummyState> {
 
+
+        private final AtomicBoolean injectFailure;
+        private final String[] failureMethods;
+
+        /**
+         * Constructs a MetaDataStateFormat object for storing/retrieving DummyState with no failure injection
+         */
         Format(String prefix) {
             super(prefix);
+            this.injectFailure = new AtomicBoolean(false);
+            this.failureMethods = new String[]{};
+        }
+
+        /**
+         * Constructs a MetaDataStateFormat object for storing/retrieving DummyState
+         *
+         * @param prefix - state file prefix
+         * @param injectFailure - whether to inject failures in Directory method calls, returned by MetaDataStateFormat.newDirectoryMethod
+         * @param failureMethods - one of these method calls will fail on Directory object if injectFailure is set to true.
+         *                       Method that will fail is randomly selected on each newDirectory method call.
+         */
+        Format(String prefix, AtomicBoolean injectFailure, String... failureMethods) {
```

Comment:
instead of injecting the AtomicBoolean, it might be nicer to just create it internally, and access it on the `Format` object

Reply:
++